### PR TITLE
fix: correctly free memory in inflation layer

### PR DESCRIPTION
**Pull request by Nisarg236**
[AB#44812](https://dev.azure.com/rapyuta-robotics/sootballs/_workitems/edit/44812)

### CODE REVIEW

**Comment by Nisarg236** [comment]
credit to the original pr in nav2: https://github.com/ros-navigation/navigation2/pull/4424

File: costmap_2d/plugins/inflation_layer.cpp [hunk 1]
```diff
@@ -231,19 +232,27 @@ void InflationLayer::updateCosts(costmap_2d::Costmap2D& master_grid, int min_i,
...
+  for (auto& dist_bin: inflation_cells_)
   {
-    for (const auto& current_cell: dist_bin)
+    dist_bin.reserve(200);
+    for (std::size_t i = 0; i < dist_bin.size(); ++i)
     {
       // process all cells at distance dist_bin.first
-      unsigned int index = current_cell.index_;
+      // unsigned int index = current_cell.index_;
+      auto& current_cell = dist_bin[i];
```
Comment (renan028) [inline costmap_2d/plugins/inflation_layer.cpp]:
```suggestion
      const auto& current_cell = dist_bin[i];
```

**Review by renan028 — APPROVED** [review]

**Comment by renan028** [comment]
nice!

**Review by corot — APPROVED** [review]
Thanks!
Just one suggestion for the next PRs: please add some context to the description, like a link to the original PR, and ideally show the improvement with a video or data

**Pull request merged**